habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ContainerizeDependenciesMojo.java

```diff
@@ -0,0 +1,156 @@
+        }
+
+        FileSetManager fileSetManager = new FileSetManager();
+        for (Path project : dependencyFileSets.keySet()) {
+            FileSet fileSet = dependencyFileSets.get(project);
+            logger.info("Copying {} files from {}",
+                    fileSetManager.getIncludedFiles(fileSet).length,
+                    project.getFileName()
+            );
+            for (String includedFile : fileSetManager.getIncludedFiles(fileSet)) {
+                Path relativePath = project.resolve(includedFile);
+                Files.createDirectories(destRoot.resolve(relativePath).getParent());
+                Files.copy(srcRoot.resolve(relativePath), destRoot.resolve(relativePath));
+            }
+        }
+    }
+
+    private FileSet getDefaultFileSet() {
+        FileSet fileSet = new FileSet();
+        fileSet.addExclude("target/**");
+        fileSet.addExclude(".venv/**");
+        fileSet.addExclude("dist/**");
+        return fileSet;
+    }
+
+    /**
+     * Checks listed habushu-type dependencies against the set of projects included in the Maven build's session
+     * @return the corresponding Maven projects that match the habushu-type dependencies
+     */
+    protected Set<MavenProject> getHabushuProjects() {
+        HashSet<MavenProject> habushuProjects = new HashSet<>();
+        collectHabushuDependenciesAsProjects(project, habushuProjects);
+        return habushuProjects;
+    }
+
+    /**
+     * Collects the projects of the habushu-type dependencies and adds them to the given project set
+     * @param currentProject the project to interrogate the habushu-type dependencies against
+     * @param habushuProjects the set to append to with matching projects
+     */
+    private void collectHabushuDependenciesAsProjects(MavenProject currentProject, Set<MavenProject> habushuProjects) {
+        Set<String> habushuDeps = currentProject.getDependencies().stream()
+                .filter(d -> HABUSHU.equals(d.getType()))
+                .map(ContainerizeDependenciesMojo::toGav)
+                .collect(Collectors.toSet());
+        for (MavenProject project : getSession().getProjects()) {
+            if (habushuDeps.contains(toGav(project))) {
+                logger.info("Found habushu-type dependency's project {}.", project);
+                habushuProjects.add(project);
+                collectHabushuDependenciesAsProjects(project, habushuProjects);
+            }
+        }
+    }
+
+    private static String toGav(Dependency dependency) {
+        return dependency.getGroupId() + ":" + dependency.getArtifactId() + ":" + dependency.getVersion();
+    }
+    private static String toGav(MavenProject project) {
+        return project.getGroupId() + ":" + project.getArtifactId() + ":" + project.getVersion();
+    }
+
+    protected MavenSession getSession() {
+        return this.session;
+    }
+
+    protected void setAnchorSourceDirectory(File anchorSourceDirectory) {
```

Comment:
Q: Is this required since we're setting the base directory in the tests??

Reply:
This was originally done for testing to bypass our method of determining the project root directory which recursively called `project.getParent()`. The `project.getParent()` doesn't work with the maven plugin testing harness as the parents cannot be set. However, now that we're using `session.getExecutionRootDirectory()` for determining the root directory, it's possible this bypass is no longer needed.

Reply:
Also possible it might still be necessary though depending on the output of `session.getExecutionRootDirectory()`

Reply:
Per previous comment, I broke testing trying to fix, will improve this for the follow-on